Gems/Atom/Feature/Common/Code/Source/SkyAtmosphere/SkyAtmosphereFeatureProcessor.cpp

```diff
@@ -0,0 +1,355 @@
+        EnableSceneNotification();
+    }
+    
+    void SkyAtmosphereFeatureProcessor::Deactivate()
+    {
+        DisableSceneNotification();
+
+        m_params.Clear();
+
+        m_skyAtmosphereParentPasses.clear();
+    }
+
+    
+    SkyAtmosphereFeatureProcessor::AtmosphereId SkyAtmosphereFeatureProcessor::CreateAtmosphere()
+    {
+        size_t index = m_params.Reserve();
+        if (index >= std::numeric_limits<AtmosphereId::IndexType>::max())
+        {
+            m_params.Release(index);
+            return AtmosphereId::Null;
+        }
+
+        AtmosphereId id = AtmosphereId(aznumeric_cast<AtmosphereId::IndexType>(index));
+        m_atmosphereIds.insert(id);
+        InitializeAtmosphere(id);
+
+        return id;
+    }
+
+    void SkyAtmosphereFeatureProcessor::ReleaseAtmosphere(AtmosphereId id)
+    {
+        if (id.IsValid())
+        {
+            m_params.Release(id.GetIndex());
+        }
+        m_atmosphereIds.erase(id);
+
+        for (auto pass : m_skyAtmosphereParentPasses )
+        {
+            pass->ReleaseAtmospherePass(id);
+        }
+
+        m_passNeedsUpdate = true;
+    }
+
+    void SkyAtmosphereFeatureProcessor::Enable(AtmosphereId id, bool enable)
+    {
+        auto& params = m_params.GetElement(id.GetIndex());
+        params.m_enabled = enable;
+        m_passNeedsUpdate = true;
+    }
+
+    bool SkyAtmosphereFeatureProcessor::IsEnabled(AtmosphereId id)
+    {
+        auto& params = m_params.GetElement(id.GetIndex());
+        return params.m_enabled;
+    }
+
+    void SkyAtmosphereFeatureProcessor::SetFastSkyEnabled(AtmosphereId id, bool enabled)
+    {
+        auto& params = m_params.GetElement(id.GetIndex());
+        params.m_fastSkyEnabled = enabled;
+        m_passNeedsUpdate = true;
+    }
+
+    void SkyAtmosphereFeatureProcessor::SetSunDirection(AtmosphereId id, const Vector3& direction)
+    {
+        auto& params = m_params.GetElement(id.GetIndex());
+        params.m_sunDirection = direction;
+        m_passNeedsUpdate = true;
+    }
+
+    void SkyAtmosphereFeatureProcessor::SetLuminanceFactor(AtmosphereId id, const AZ::Vector3& factor)
+    {
+        auto& params = m_params.GetElement(id.GetIndex());
+        params.m_luminanceFactor = factor;
+        params.m_lutUpdateRequired = true;
+        m_passNeedsUpdate = true;
+    }
+
+    void SkyAtmosphereFeatureProcessor::SetMinMaxSamples(AtmosphereId id, uint32_t minSamples, uint32_t maxSamples)
+    {
+        auto& params = m_params.GetElement(id.GetIndex());
+        params.m_minSamples = minSamples;
+        params.m_maxSamples = maxSamples;
+        m_passNeedsUpdate = true;
+    }
+
+    void SkyAtmosphereFeatureProcessor::SetRayleighScattering(AtmosphereId id, const AZ::Vector3& scattering)
+    {
+        auto& params = m_params.GetElement(id.GetIndex());
+        params.m_rayleighScattering = scattering;
+        params.m_lutUpdateRequired = true;
+        m_passNeedsUpdate = true;
+    }
+
```

Comment:
Seems like a lot of code repetition. Could we not have 2-3 generic methods that will update all parameters and mark pass or parameters as dirty? it's no big overhead but much less per parameter code.